### PR TITLE
Addon for REFramework.

### PR DESCRIPTION
**Pull request by youwereeatenbyalid**
Adds support for [REFramework](https://github.com/praydog/REFramework/), A mod framework, scripting platform, and modding tool for RE Engine games. 

Specifically, the addon provides basic autocomplete support for the lua scripting system (documented [here](https://cursey.github.io/reframework-book/index.html).)

### CODE REVIEW

**Comment by youwereeatenbyalid** [comment]
Was getting tired of cross referencing the wiki page for this project, so I wrote this up. First time doing anything with the add-on system but I think I got it all right.

File: .gitmodules [hunk 1]
```diff
@@ -122,3 +122,6 @@
 [submodule "addons/minetest/module"]
 	path = addons/minetest/module
 	url = https://git.sr.ht/~fgaz/minetest-lls-definitions
+[submodule "addons/REFramework-LLS/module"]
+	path = addons/REFramework-LLS/module
+	url = https://github.com/youwereeatenbyalid/REFramework-LLS
```
Comment (carsakiller) [inline .gitmodules:127]: The `url` should be a git clone URL, not the HTML page, like the below:
```suggestion
	url = https://github.com/youwereeatenbyalid/REFramework-LLS.git
```

Did you add the submodule using the following command?
`git submodule add https://github.com/youwereeatenbyalid/REFramework-LLS.git addons/REFramework-LLS/module`

Comment (youwereeatenbyalid) [reply .gitmodules]: I did not, wrote  `git submodule add https://github.com/youwereeatenbyalid/REFramework-LLS addons/REFramework-LLS/module` I'm pretty certain. Didn't realize there was a difference between the two.

**Comment by carsakiller** [comment]
Well done, thanks!

**Pull request merged**